pyomo/contrib/doe/fim_doe.py

```diff
@@ -557,7 +575,7 @@ def optimize_doe(self,  design_values, if_optimize=True, objective_option='det',
             return analysis_square
 
     def compute_FIM(self, design_values, mode='sequential_finite', FIM_store_name=None, specified_prior=None,
```

Comment:
This is a really big function. Does it make sense to divide it into a sequence of smaller private methods?

Reply:
How much of the workflow in this function is general across all supported modes? Or is most of this function series of if statements that have different sections for each mode? If the latter, we should strongly consider moving the code for each mode into a private method in the class. But let's do this AFTER identifying common code across the modes that can be generalized.